Added yum retention script.

bpankowski opened the pull request.

bpankowski committed:
Added yum retention script.

bpankowski committed:
Code cleanup.

bpankowski committed:
Delete of retention date check

bpankowski committed:
Delete of retention date check

kravvcuCL commented on tungsten_ci_utils/yum_cleanup/rpm_cleanup.groovy:
```diff
@@ -0,0 +1,80 @@
+def retentionCount = 25;
+def tagList = [];
+def repositoryName = 'yum-tungsten';
+def whitelisted_tag_suffixes = ["queens", "ocata", "newton", "latest", "40", "94", "122", "129", "161", "214", "309", "360"].toArray();
```
remove queens ocata newton latest from this list. This only appplies to docker image taggin scheme.

kravvcuCL commented on tungsten_ci_utils/yum_cleanup/rpm_cleanup.groovy:
```diff
@@ -0,0 +1,80 @@
+    int deletedComponentCount = 0;
+    def listOfComponents = ImmutableList.copyOf(components);
+
+// comp.version() expected to be in a format {{ os }}-{{ realse }}-{{ branch }}-{{ build }} e.g. rhel-queens-master-386 
```
Adjust this. The scheme for RPM naming is name-branch-build_number.architecture.rpm
Sometimes 'el6' or 'el7' is added between build-number and achitecture like you've noticed.

bpankowski committed:
Code cleanup. Change of comments due to different version structure.

bpankowski committed:
Merge branch 'yum-cleanup' of https://github.com/tungsten-infra/ci-utils into yum-cleanup

bpankowski committed:
Roleback of retention date

bpankowski committed:
Changed comment about comp.version()

bpankowski committed:
Add new whitelist of tags for yum-repository.

Add whitelist with branch-buildnumber structure.
Fix of minor bug with retention date which is not necessary.

bpankowski committed:
Fix of wrong description.

Change of line number in description.

jluk-codi commented on tungsten_ci_utils/yum_cleanup/rpm_cleanup.groovy:
```diff
@@ -0,0 +1,93 @@
+// Script created for retention of private docker repo on Nexus.
```
Change Docker to YUM

jluk-codi commented on tungsten_ci_utils/yum_cleanup/rpm_cleanup.groovy:
```diff
@@ -0,0 +1,93 @@
+
+def retentionCount = 25;
+def tagList = [];
+def repositoryName = 'yum-tungsten';
```
Change this to yum-tungsten-nightly

bpankowski committed:
Change of repositoryName and description.

Change value of repositoryName to 'yum-tungsten-nightly'.
Change of script description.

bpankowski committed:
Add of new feature to delete empty repos with repodata only.

That allows to erase empty folders from Nexus yum repository.

bpankowski committed:
Add of new feature to delete empty repos with repodata only.

That allows to erase empty folders from Nexus yum repository.

bpankowski committed:
Merge branch 'yum-cleanup' of https://github.com/tungsten-infra/ci-utils into yum-cleanup

bpankowski committed:
Revert "Change of repositoryName and description."

This reverts commit 8fb0906ba14f8935a5e59faf6100010ca38f63bd.

bpankowski committed:
Revert "Revert "Change of repositoryName and description.""

This reverts commit 15d86feb40e3ad0aa9c6b195366a5e721a52f2b7.

bpankowski committed:
Revert "Add of new feature to delete empty repos with repodata only."

This reverts commit 38b0856c776e75cd64e37583b3dfba58e5518678.

kravvcuCL commented on tungsten_ci_utils/yum_cleanup/rpm_cleanup.groovy:
```diff
@@ -0,0 +1,93 @@
+        } else if (whitelisted_tag_suffixes.contains(two_parts_build_number)){
+            log.info("Component skipped: ${comp.name()} ${comp.version()}");
+        }else {
+            if (tagList.count(build_number.toInteger()) == 0) {
```
If you want to check whether an object is part of a list use the 'contains' method

https://docs.oracle.com/javase/7/docs/api/java/util/List.html#contains(java.lang.Object)

kravvcuCL commented on tungsten_ci_utils/yum_cleanup/rpm_cleanup.groovy:
```diff
@@ -0,0 +1,93 @@
+        }else {
+            if (tagList.count(build_number.toInteger()) == 0) {
+                tagList.add(build_number.toInteger());
+                println tagList.sort();
```
Maybe print after the iteration finishes? outside the 'each' closure? Or is this intended and you want to print the whole list when it grows?

kravvcuCL commented on tungsten_ci_utils/yum_cleanup/rpm_cleanup.groovy:
```diff
@@ -0,0 +1,93 @@
+        }
+        def two_parts_build_number = tagSplited[tagSplited.length - 2] + "-" + build_number;
+        log.info("new two parts ${two_parts_build_number} vs ${build_number} vs ${tagSplited}");
+        if(tagList.size() > retentionCount){
```
Think you can move this outside, before the 'each' closure as this if/else should be run only once, to either trim the list or just return

kravvcuCL commented on tungsten_ci_utils/yum_cleanup/rpm_cleanup.groovy:
```diff
@@ -0,0 +1,93 @@
+        if(!build_number.isNumber()){
+            build_number = build_number.split("\\.")[0];
+        }
+        def two_parts_build_number = tagSplited[tagSplited.length - 2] + "-" + build_number;
```
Please add some comment on what we try to cover here, an example of a two part build number

bpankowski committed:
Batch of smaller fixes.

Add of documentation about variables to the script.
Delete of log with tagList used for debug.

bpankowski committed:
Move of if statment outside of each closure.

First check is done on the size of tagList.

kravvcuCL commented on tungsten_ci_utils/yum_cleanup/rpm_cleanup.groovy:
```diff
@@ -0,0 +1,99 @@
+        }
+        def two_parts_build_number = tagSplited[tagSplited.length - 2] + "-" + build_number;
+        log.info("new two parts ${two_parts_build_number} vs ${build_number} vs ${tagSplited}");
+        if(tagList.size() > retentionCount){
```
something went wrong here, you're checking the same condition in an 'if' block which already checked it.
